service/src/java/org/apache/hive/service/cli/thrift/ThriftHttpServlet.java

```diff
@@ -214,9 +214,13 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
             }
           } else if (HiveSamlUtils.isSamlAuthMode(authType)) {
             // check if this request needs a SAML redirect
-            if (needsRedirect(request, response)) {
+            String authHeader = request.getHeader(HttpAuthUtils.AUTHORIZATION);
+            if ((authHeader == null || authHeader.isEmpty()) && needsRedirect(request, response)) {
               doSamlRedirect(request, response);
               return;
+            } else if(authHeader.toLowerCase().startsWith(HttpAuthUtils.BASIC.toLowerCase())) {
+              //LDAP Authentication if the header starts with Basic
+              clientUserName = doPasswdAuth(request, HiveAuthConstants.AuthTypes.NONE.toString());
```

Comment:
Do we need to handle the case that `HIVE_SERVER2_TRUSTED_PROXY_TRUSTHEADER` is present?

Reply:
so HIVE_SERVER2_TRUSTED_PROXY_TRUSTHEADER is for scenarios when the authentication is performed at a boundary service like knox/sidecars etc. After the auth is performed, the authenticator will add this TRUST header into the request and along with the username and forward the request to the service.
HS2 will trust that the authentication has already been done and relies on it.

Reply:
@hsnusonic to answer your question, yes. This should be the first check in the authentication process. We dont expect this trust header to exist in many scenarios. Sai will be looking to rearrange the code upstream.